### PR TITLE
Puzzle CRUD: Bug fixes

### PR DESCRIPTION
**Pull request by amoliksingh**
Loom Link: Wed Mar 9 Bug Bash
https://www.notion.so/uwblueprintexecs/9240fcde550d4b2aa4d3e4fd3a78766d?v=d94b8666093d47f7bab9eb6323260a0c&p=93bf13213cc64ea0b1ff940b7827a4cd
image
image
image

### CODE REVIEW

**Review by ahmedhamodi — CHANGES_REQUESTED** [review]
@jaydhulia @drbalar there is an issue here. If we make ordered puzzles dynamic, the logic on showing the placeholder tiles for the cards on backend side needs to be updated. Watch Amolik's video to understand what I mean.

I think we should keep it fixed at 5 until we update the backend logic to show placeholder values properly for n-image ordered puzzles.

**Comment by jaydhulia** [comment]
Yeah, I guess we need defaults on each possible number (2, 3, 4, 5) and update it accordingly on the ordered puzzle component in Aframe. Me or @drbalar will work on that piece tomorrow.

File: src/Components/Admin/ObjectEditor/unorderedpuzzle.js [hunk 1]
```diff
@@ -133,10 +127,10 @@ export default function UnorderedPuzzle(props) {
...
 
         const reorderedList = reorder(sourceIndex, destinationIndex);
-        const temp = reorderedList[sourceIndex].xTarget;
-        reorderedList[sourceIndex].xTarget =
-            reorderedList[destinationIndex].xTarget;
-        reorderedList[destinationIndex].xTarget = temp;
+        // const temp = reorderedList[sourceIndex].xTarget;
+        // reorderedList[sourceIndex].xTarget =
+        //     reorderedList[destinationIndex].xTarget;
+        // reorderedList[destinationIndex].xTarget = temp;
```
Comment (ahmedhamodi) [inline src/Components/Admin/ObjectEditor/unorderedpuzzle.js:133]: Intentional?

File: src/Components/Admin/ObjectEditor/unorderedpuzzle.js [hunk 2]
```diff
@@ -150,36 +144,37 @@ export default function UnorderedPuzzle(props) {
...
 
     const deleteImage = (index) => {
-        const tempImages = JSON.parse(JSON.stringify(images));
+        const tempImages = images;
         tempImages.splice(index, 1);
+        props.saveImages(tempImages);
         setImages(tempImages);
-        props.deleteImage(index);
         setUploaded(false);
     };
```
Comment (ahmedhamodi) [inline src/Components/Admin/ObjectEditor/unorderedpuzzle.js:161]: Can you explain this logic, intuitively the "deleteImage" function should be calling props.deleteImage haha, but I don't know the actual logic going on behind the scenes here.

**Review by ahmedhamodi — COMMENTED** [review]
Initial pass. Some things I wasn't too sure on

File: src/Components/Admin/ObjectEditor/unorderedpuzzle.js (hunk 2 above)
Comment (amoliksingh) [reply src/Components/Admin/ObjectEditor/unorderedpuzzle.js:161]: Yea, so the JSON.parse(JSON.stringify()) logic messes up the byte array attribute, so I switched it back to just = images, and the props.saveImages(tempImages) replaces the deleteImage line

**Review by ahmedhamodi — APPROVED** [review]
LGTM, demoed to Dhruvin and I over call.

**Pull request merged**